app/javascript/packages/react-hooks/index.ts

```diff
@@ -1,2 +1,3 @@
 export { default as useDidUpdateEffect } from './use-did-update-effect';
 export { default as useIfStillMounted } from './use-if-still-mounted';
+export { default as useInstanceId } from './use-instance-id';
```

Comment:
Note: React 18 now includes [a hook designed for this purpose](https://reactjs.org/docs/hooks-reference.html#useid), but we're currently blocked from upgrading due to Internet Explorer support.

See: #6160